src/tlo/methods/hsi_generic_first_appts.py

```diff
@@ -6,369 +6,179 @@
 the onset of symptoms. Non-emergency symptoms lead to `HSI_GenericFirstApptAtFacilityLevel0` and emergency symptoms
 lead to `HSI_GenericEmergencyFirstApptAtFacilityLevel1`.
 """
-import pandas as pd
+from collections import namedtuple
+from typing import TYPE_CHECKING, Any, Literal, OrderedDict
 
 from tlo import logging
 from tlo.events import IndividualScopeEventMixin
-from tlo.methods.bladder_cancer import (
-    HSI_BladderCancer_Investigation_Following_Blood_Urine,
-    HSI_BladderCancer_Investigation_Following_pelvic_pain,
-)
-from tlo.methods.breast_cancer import (
-    HSI_BreastCancer_Investigation_Following_breast_lump_discernible,
-)
-from tlo.methods.care_of_women_during_pregnancy import (
-    HSI_CareOfWomenDuringPregnancy_PostAbortionCaseManagement,
-    HSI_CareOfWomenDuringPregnancy_TreatmentForEctopicPregnancy,
-)
-from tlo.methods.chronicsyndrome import HSI_ChronicSyndrome_SeeksEmergencyCareAndGetsTreatment
-from tlo.methods.epilepsy import HSI_Epilepsy_Start_Anti_Epileptic
-from tlo.methods.hiv import HSI_Hiv_TestAndRefer
 from tlo.methods.hsi_event import HSI_Event
-from tlo.methods.labour import HSI_Labour_ReceivesSkilledBirthAttendanceDuringLabour
-from tlo.methods.measles import HSI_Measles_Treatment
-from tlo.methods.mockitis import HSI_Mockitis_PresentsForCareWithSevereSymptoms
-from tlo.methods.oesophagealcancer import HSI_OesophagealCancer_Investigation_Following_Dysphagia
-from tlo.methods.other_adult_cancers import (
-    HSI_OtherAdultCancer_Investigation_Following_early_other_adult_ca_symptom,
-)
-from tlo.methods.prostate_cancer import (
-    HSI_ProstateCancer_Investigation_Following_Pelvic_Pain,
-    HSI_ProstateCancer_Investigation_Following_Urinary_Symptoms,
-)
+
+if TYPE_CHECKING:
+    from tlo import Module
 
 logger = logging.getLogger(__name__)
 logger.setLevel(logging.INFO)
 
+class HSI_BaseGenericFirstAppt(HSI_Event, IndividualScopeEventMixin):
+    """
+    """
+    MODULE_METHOD_ON_APPLY: Literal[
+        "do_at_generic_first_appt", "do_at_generic_first_appt_emergency"
+    ]
+
+    def __init__(self, module, person_id) -> None:
+        super().__init__(module, person_id=person_id)
+        # No footprint, as this HSI (mostly just) determines which
+        # further HSI will be needed for this person. In some cases,
+        # small bits of care are provided (e.g. a diagnosis, or the
+        # provision of inhaler).
+        self.EXPECTED_APPT_FOOTPRINT = self.make_appt_footprint(
+            {}
+        )
+
+    def _diagnosis_fn(self, tests, use_dict: bool = False, report_tried: bool = False) -> Any:
```

Comment:
Could the return type hint be more specific here as docstring suggests it will be a dict of some form?